openapi.yml

```diff
@@ -730,6 +769,121 @@ components:
+        document_id:
+          <<: *non-empty-string
+        id:
+          type: integer
+        requester:
+          $ref: "#/components/schemas/OrganizationDelegate"
+        status:
+          $ref: '#/components/schemas/RequestStatus'
+        type:
+          type: string
+          x-extensible-enum:
+            - "USER_DELEGATION"
+      required:
+        - document_id
+        - id
+        - organization
+        - requester
+        - status
+        - type
+      example:
+        {
+          document_id: "document-id",
+          id: 12345,
+          organization: {
+            "fiscal_code": "86000470830",
+            "ipa_code": "c_e043",
+            "legal_representative": {
+              "family_name": "Spano'",
+              "given_name": "Ignazio Alfonso",
+              "fiscal_code": "BCDFGH12A21Z123D",
+              "phone_number": "5550000000",
+              "email": "indirizzo00@email.pec.it",
+              "role": "ORG_MANAGER"
+            },
+            "name": "Comune di Gioiosa Marea",
+            "pec": "indirizzo00@email.pec.it",
+            "registration_status": "PRE_DRAFT",
+            "scope": "NATIONAL"
+          },
+          requester: {
+            "email": "delegate@example.com",
+            "family_name": "Rossi",
+            "fiscal_code": "RSSMTT77S12A123D",
+            "given_name": "Matteo",
+            "role": "ORG_DELEGATE"
+          },
+          status: "SUBMITTED",
+          type: "USER_DELEGATION"
+        }
+    OrganizationRegistrationRequest:
+      type: object
+      properties:
+        organization:
+          $ref: "#/components/schemas/OrganizationToBeRegistered"
+        document_id:
+          <<: *non-empty-string
+        id:
+          type: integer
+        requester:
+          $ref: "#/components/schemas/OrganizationDelegate"
+        status:
+          $ref: '#/components/schemas/RequestStatus'
+        type:
+          type: string
+          x-extensible-enum:
+            - "ORGANIZATION_REGISTRATION"
+      required:
+        - document_id
+        - id
+        - organization
+        - status
+        - type
+      example:
+        {
+          document_id: "document-id",
+          id: 12345,
+          organization: {
+            "fiscal_code": "86000470830",
+            "ipa_code": "c_e043",
+            "legal_representative": {
+              "family_name": "Spano'",
+              "given_name": "Ignazio Alfonso",
+              "fiscal_code": "BCDFGH12A21Z123D",
+              "phone_number": "5550000000",
+              "email": "indirizzo00@email.pec.it",
+              "role": "ORG_MANAGER"
+            },
+            "name": "Comune di Gioiosa Marea",
+            "pec": "indirizzo00@email.pec.it",
+            "registration_status": "PRE_DRAFT",
+            "scope": "NATIONAL"
+          },
+          status: "SUBMITTED",
+          type: "USER_DELEGATION"
+        }
+    Request:
+      oneOf:
```

Comment:
check the generated type it must be a `t.union`

Reply:
It is:
```typescript
export const Request = t.union(
  [OrganizationRegistrationRequest, UserDelegationRequest],
  "Request"
);
```